### PR TITLE
Testing React with Jest

### PR DESCRIPTION
**Pull request by ad9311**
### Greetings! 👋🏽 

In this PR we have added testing units for the React Math Magicians app.

This changes include:

- Unit tests for calculate.js and operate.js.
- Unit tests for the Calculator.js and Key.js.
- Unit tests for the pages; Home.js, Calculator.js and Quote.js.

### CODE REVIEW

File: src/__tests__/logic.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    obj = calculate(obj, '3');
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual('4');
+  });
+
+  test('6 - 2', () => {
+    const total = operate('6', '2', '-');
+    expect(total).toEqual('4');
+  });
+});
```
Comment (george-shammar) [inline src/__tests__/logic.test.js:18]: - [ ] Please, all the functionalities & scenarios of the` operate.js` and `calculate.js` modules should be covered using Jest. Here, you've only tested `minus and division` operators.

File: src/__tests__/__snapshots__/pages.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+exports[`Test routes test home 1`] = `
+<h2
+  class="fs-4"
+>
+  Welcome to our page!
+</h2>
+`;
+
+exports[`Test routes test quotes 1`] = `
+<strong>
```
Comment (george-shammar) [inline src/__tests__/__snapshots__/pages.test.js.snap:20]: - [ ] Kindly add snapshot tests for the navbar component as well. There should be the usage of Jest snapshots for testing all the components.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!

:heavy_check_mark: This is truly impressive. :clap: 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ad9311** [comment]
Following the previous reviewer instructions, we have made the following changes:

- Add more test units for calculation.js and operation.js. These includes all the mathematical operations.
- Add snapshot test to Navbar using a mock.

**Review by george-shammar — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**